File: app/src/main/java/com/muratcan/apps/petvaccinetracker/widget/WidgetUpdateService.java
package com.muratcan.apps.petvaccinetracker.widget;

import android.app.IntentService;
import android.appwidget.AppWidgetManager;
import android.content.ComponentName;
import android.content.Intent;

public class WidgetUpdateService extends IntentService {
    private static final String ACTION_UPDATE_WIDGET = "com.muratcan.apps.petvaccinetracker.UPDATE_WIDGET";

    public WidgetUpdateService() {
        super("WidgetUpdateService");
    }

    public static Intent createUpdateIntent(android.content.Context context) {
        Intent intent = new Intent(context, WidgetUpdateService.class);
        intent.setAction(ACTION_UPDATE_WIDGET);
        return intent;
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        if (ACTION_UPDATE_WIDGET.equals(intent.getAction())) {
            AppWidgetManager appWidgetManager = AppWidgetManager.getInstance(this);
            ComponentName componentName = new ComponentName(this, VaccineWidgetProvider.class);
            int[] appWidgetIds = appWidgetManager.getAppWidgetIds(componentName);
            
            VaccineWidgetProvider widgetProvider = new VaccineWidgetProvider();
            widgetProvider.onUpdate(this, appWidgetManager, appWidgetIds);
        }
    }
} 

File: app/src/main/java/com/muratcan/apps/petvaccinetracker/widget/WidgetUpdateService.kt
package com.muratcan.apps.petvaccinetracker.widget

import android.app.IntentService
import android.appwidget.AppWidgetManager
import android.content.ComponentName
import android.content.Context
import android.content.Intent

class WidgetUpdateService : IntentService("WidgetUpdateService") {
    companion object {
        private const val ACTION_UPDATE_WIDGET = "com.muratcan.apps.petvaccinetracker.UPDATE_WIDGET"

        fun updateWidget(context: Context) {
            val intent = Intent(context, WidgetUpdateService::class.java)
            context.startService(intent)
        }
    }

    override fun onHandleIntent(intent: Intent?) {
        if (ACTION_UPDATE_WIDGET == intent?.action) {
            val appWidgetManager = AppWidgetManager.getInstance(this)
            val componentName = ComponentName(this, VaccineWidgetProvider::class.java)
            val appWidgetIds = appWidgetManager.getAppWidgetIds(componentName)

            val widgetProvider = VaccineWidgetProvider()
            widgetProvider.onUpdate(this, appWidgetManager, appWidgetIds)
        }
    }
} 